contracts/Factory.vy

```diff
@@ -427,6 +453,62 @@ def deploy_metapool(
     return pool
 
 
+@external
+def add_existing_pools(_pools: address[N_POOLS], _base_pool: address) -> bool:
+    """
+    @notice Add existing factory pools to this factory
+    @dev Base pools that are used by the pools to be added must
+        be added separately with `add_base_pool`. All pools given
+        in `_pools` must have the same `_base_pool`
+    @param _pools Addresses of existing pools to add
+    @param _base_pool Address of the base pool for `_pools`
+    """
+    assert msg.sender == self.admin  # dev: admin-only function
+    assert self.base_pool_data[_base_pool].coins[0] != ZERO_ADDRESS # dev: base pool does not exist
+
+    registry: address = AddressProvider(ADDRESS_PROVIDER).get_registry()
+    base_lp_token: address = Registry(registry).get_lp_token(_base_pool)
+
+    for pool in _pools:
+        if pool == ZERO_ADDRESS:
+            break
+
+        # add pool to pool list
+        length: uint256 = self.pool_count
+        self.pool_list[length] = pool
+        self.pool_count = length + 1
```

Comment:
Can we track `length` in memory and only read from / write to `self.pool_count` once at the end?